hypervisor/dm/vpci/pci_pt.c

```diff
@@ -120,11 +58,11 @@ static void vdev_pt_unmap_mem_vbar(struct pci_vdev *vdev, uint32_t idx)
 
 	vbar = &vdev->bar[idx];
 
-	if (vdev->bar_base_mapped[idx] != 0UL) {
+	if (vbar->base != 0UL) {
 		ept_del_mr(vm, (uint64_t *)(vm->arch_vm.nworld_eptp),
-			vdev->bar_base_mapped[idx], /* GPA (old vbar) */
+			vbar->base, /* GPA (old vbar) */
 			vbar->size);
-		vdev->bar_base_mapped[idx] = 0UL;
+		vbar->base = 0UL;
```

Comment:
Unmap is unmap. It is irrelevant to change vbar->base.  Same for the 1st patch and many places in this patch.